### PR TITLE
feat: refactored backend unit tests

### PR DESCRIPTION
**Pull request by NithinKuruba**
1. Made use of supertest
2. Generate code coverage and html report
3. developed latest common components to easily create integrations and cleanup database

### CODE REVIEW

File: lambda/__tests__/helpers/server.ts [hunk 1]
```diff
@@ -14,13 +16,15 @@ const tryJSON = (str) => {
...
+const appRouter = express.Router();
+const actionsRouter = express.Router();
+const apiRouter = express.Router();
 app.use(bodyParser.json());
 app.use(bodyParser.urlencoded({ extended: false }));
 app.set('etag', false);
 app.disable('x-powered-by');
 
-router.use(
+app.use(
```
Comment (thegentlemanphysicist) [inline lambda/__tests__/helpers/server.ts:27]: Why the change here?  Are the methods express.Router().use  and express().use the same funtionality wise?

Comment (NithinKuruba) [reply lambda/__tests__/helpers/server.ts:27]: @thegentlemanphysicist that's just a rename and functionality does not change with it

File: lambda/css-api/src/routes.ts [hunk 2]
```diff
@@ -60,6 +60,8 @@ export const setRoutes = (app: any) => {
 
   app.use(async (req, res, next) => {
     const auth: Auth = await authenticate(req.headers);
+    console.log(auth);
```
[the remark this answers is not in the document]
Comment (NithinKuruba) [reply lambda/css-api/src/routes.ts]: Removed it as I used it for debugging

**Review by thegentlemanphysicist — DISMISSED** [review]

**Review by thegentlemanphysicist — APPROVED** [review]
